packages/components/src/presentation-components/properties/inputs/UniquePropertyValuesSelector.tsx

```diff
@@ -87,24 +89,24 @@ export function UniquePropertyValuesSelector(props: UniquePropertyValuesSelector
   };
 
   const isOptionSelected = (option: UniqueValue): boolean => selectedValues.map((selectedValue) => selectedValue.displayValue).includes(option.displayValue);
-
   const ruleset = useUniquePropertyValuesRuleset(descriptor.ruleset, field);
-  const loadTargets = useUniquePropertyValuesLoader({ imodel, ruleset, field, descriptorInputKeys });
+  const loadValues = useUniquePropertyValuesLoader({ imodel, property, descriptor, ruleset, field, descriptorInputKeys });
 
   return (
     <AsyncSelect
       value={selectedValues}
-      loadOptions={async (_, options) => loadTargets(options.length)}
+      loadOptions={async (_, options) => loadValues(searchInput, options.length)}
```

Comment:
Shouldn't we use first argument of `loadOptions`? Also maybe it's worth debounce loading options while typing in selector? `AsyncSelect` supports passing debounce time if I remember correctly.

Reply:
Should we? I am using the `searchInput` instead of the passed value just because its already stored in lowercase. From my testing it seems like the state is always updated before the `loadOptions` call.

Reply:
I think using search value passed to `loadOptions` are correct approach especially if we enable debounce. From our discussion I understood that `searchInput` state is mainly for refreshing cache inside `AsyncSelect`.

Reply:
Yes the purpose of `searchInput` is for refreshing the cache. Switched to using the passed value and added debounce timeout.